include/eve/concept/element_type.hpp

```diff
@@ -0,0 +1,20 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/simd.hpp>
+#include <eve/traits/element_type.hpp>
+
+namespace eve
+{
+  template<typename T, typename... Ts>
+  concept same_element_type = (std::same_as<element_type_t<T>, element_type_t<Ts>> || ...);
+
+  template<typename T, typename... Ts>
+  concept same_element_type_or_scalar = (scalar_value<T> || ... || scalar_value<Ts>) || same_element_type<T, Ts...>;
```

Comment:
This one is definitely not what you want. 
And the concept file name `element_type` is not ideal, you won't be able to guess where it is.

Let's think. 

On the surface i want to say: `same_or_scalar`. 

And then your call looks like: 

```
template <arithmetic_value T, arithmetic_value U>
  requires same_or_scalar<T, U>
```

But I think there is also some merit in `compatible_arithmetic_values`.
Then, in the future, when we talk about translations, we can add those here as well. 

What do you think?